### ISSUE TITLE
When I try to run query from plugin, it gives me "Metric reauest error"

### ISSUE THREAD

**Issue by Tugduall** [tvarit-foggy/grafana-s3-plugin#10]
Here is the configuration of the request and the result

![Screenshot from 2020-11-18 10-13-11](https://user-images.githubusercontent.com/23258388/99510165-03711b00-2987-11eb-9233-1e220f54229d.png)

![Screenshot from 2020-11-18 10-13-37](https://user-images.githubusercontent.com/23258388/99510214-12f06400-2987-11eb-9a88-913c10f05451.png)

Is it necessary to use a specific identation for the csv? or what's wrong?

Thank you in advance

**Comment by vabatox** [comment]
I get the same error. I successfully connect to bucket. copy in the key for the csv file I want to query.

{
  "request": {
    "url": "api/ds/query",
    "method": "POST",
    "data": {
      "queries": [
        {
          "compression": "NONE",
          "csv_allow_quoted_record_delimiter": false,
          "csv_comments": "#",
          "csv_field_delimiter": ",",
          "csv_file_header_info": "USE",
          "csv_quote_character": "\"",
          "csv_quote_escape_character": "\"",
          "csv_record_delimiter": "\n",
          "format": "CSV",
          "json_time_bucket": 1000000,
          "json_time_field": "",
          "json_time_field_enable": false,
          "json_time_month_first": false,
          "json_type": "DOCUMENT",
          "path": "data_uploads/prh_global_uk_sandbox/br_misc_snowflake_upload/bryan12.csv",
          "query": "SELECT * FROM s3object",
          "refId": "A",
          "datasource": "AWS S3",
          "datasourceId": 1,
          "intervalMs": 30000,
          "maxDataPoints": 598,
          "orgId": 1
        }
      ],
      "range": {
        "from": "2020-12-10T09:35:30.517Z",
        "to": "2020-12-10T15:35:30.517Z",
        "raw": {
          "from": "now-6h",
          "to": "now"
        }
      },
      "from": "1607592930517",
      "to": "1607614530517"
    },
    "hideFromInspector": false
  },
  "response": {
    "error": "rpc error: code = Unavailable desc = transport is closing",
    "message": "Metric request error"
  }
}

**Comment by burningalchemist** [comment]
@vabatox @Tugduall Hi guys, if you're still interested, I've figured out what was the issue and happy to share. :)

**Comment by vabatox** [comment]
> @vabatox @Tugduall Hi guys, if you're still interested, I've figured out what was the issue and happy to share. :)

Yes please, just started looking at this again

**Comment by Tugduall** [comment]
> @vabatox @Tugduall Hi guys, if you're still interested, I've figured out what was the issue and happy to share. :)

Yes please, I avoided the problem for now but I would like to use this pulign

**Comment by burningalchemist** [comment]
@vabatox @Tugduall The issue comes from the error handling line ([here](https://github.com/tvarit-foggy/grafana-s3-plugin/blob/master/pkg/s3-select.go#L189)). If you either remove/comment it out or wrap it properly, this should do the trick at least for handling CSVs.👍

I've also updated dependencies to include arm64 builds and added the profile field in the configuration for shared credentials file (~/.aws/credentials) for multi-account configurations. I'm going to push the release archive soon if you can't or don't want to build it locally. Hope this helps.😃

**Comment by vabatox** [comment]
Thanks @burningalchemist. Made the changes to s3-select.go & restarted docker container. Still getting same error though.

**Comment by burningalchemist** [comment]
@vabatox Oh no. 😢 Did you rebuild the whole thing? Could you also share your Dockerfile?
I'm happy to help, not sure if we can proceed here. Feel free to contact me via burning.alchemist@gmail.com 😃

**Comment by burningalchemist** [comment]
@vabatox Try the zip out - https://github.com/burningalchemist/grafana-s3-plugin/releases/tag/v1.0.1

**Comment by vabatox** [comment]
> @vabatox Try the zip out - https://github.com/burningalchemist/grafana-s3-plugin/releases/tag/v1.0.1

Thanks I've got it working. Great stuff @burningalchemist 😄

**Comment by burningalchemist** [comment]
@vabatox Great! I'm happy it works for you! 👍

**Comment by starkblaze01** [comment]
Thanks for reporting and the issue and also the fixes :)
